server/src/main/java/org/elasticsearch/search/internal/ContextIndexSearcher.java

```diff
@@ -118,6 +119,15 @@ public void removeQueryCancellation(Runnable action) {
         this.cancellable.remove(action);
     }
 
+    @Override
+    public void close() {
```

Comment:
Can you add a comment on why we're clearing the list on close ? It's easy to forget and/or to remove if we don't explain the possible leak in the reader.